### PR TITLE
91198 avs param get appts

### PR DESCRIPTION
**Pull request by cferris32**
## Summary

- This work is to add an 'avs' param to the appointments request flow so that consumers can specify if they want the response to include data retrieved from the avs service, thus bypassing these calls when we don't need them

## Related issue(s)

- https://app.zenhub.com/workspaces/appointments-team-603fdef281af6500110a1691/issues/gh/department-of-veterans-affairs/va.gov-team/91198

## Testing done

- updated unit tests and ran regression tests

### CODE REVIEW

File: modules/vaos/app/services/vaos/v2/appointments_service.rb [hunk 1]
```diff
@@ -398,8 +399,7 @@ def get_avs_link(appt)
       #
       # @return [nil] This method does not explicitly return a value. It modifies the `appt`.
       def fetch_avs_and_update_appt_body(appt)
-        # Testing AVS empty state using the below id - remove after testing is complete
-        if appt[:id] == AVS_APPT_TEST_ID
```
Comment (JunTaoLuo) [inline modules/vaos/app/services/vaos/v2/appointments_service.rb:402]: Let's remove the constant as well.

File: modules/vaos/app/services/vaos/v2/appointments_service.rb [hunk 2]
```diff
@@ -426,10 +426,10 @@ def cannot_be_cancelled?(appointment)
       # @param appt [Hash] the appointment to check
       # @return [Boolean] true if the appointment is eligible, false otherwise
       #
-      def avs_applicable?(appt)
-        return false if appt.nil? || appt[:status].nil? || appt[:start].nil?
+      def avs_applicable?(appt, avs)
+        return false if appt.nil? || appt[:status].nil? || appt[:start].nil? || avs.nil?
 
-        appt[:status] == 'booked' && appt[:start].to_datetime.past?
+        appt[:status] == 'booked' && appt[:start].to_datetime.past? && avs.to_s == 'true'
```
Comment (JunTaoLuo) [inline modules/vaos/app/services/vaos/v2/appointments_service.rb:432]: Hmm isn't avs passed in as a bool? Is the conversion to string necessary?

**Review by JunTaoLuo — COMMENTED** [review]
Mostly looks good, just had one question and one nitpick

**Comment by cferris32** [comment]
Closing this PR since we plan to rework this implementation

**Pull request closed without merging**